src/main/scala/higherkindness/compendium/http/QueryParams.scala

```diff
@@ -38,4 +38,13 @@ object QueryParams {
     )("ProtocolVersion")
 
   object ProtoVersion extends OptionalValidatingQueryParamDecoderMatcher[ProtocolVersion]("version")
+
+  implicit val validationDecoderQueryParam: QueryParamDecoder[ValidationBool] =
+    QueryParamDecoder.fromUnsafeCast[ValidationBool](qp => ValidationBool.withName(qp.value))(
+      "ValidationBool"
+    )
+
+  object ValidationParam
+      extends OptionalValidatingQueryParamDecoderMatcher[ValidationBool]("validation")
```

Comment:
Since you're providing an implicit `QueryParamDecoder[ValidationBool]`, you can replace this by:
```suggestion
  object ValidationParam 
    extends OptionalQueryParamDecoderMatcher[ValidationBool]("validation")
```

If you go to the `OptionalQueryParamDecoderMatcher` implementation, it requires a `QueryParamDecoder` implicit and it'll give you the optional value directly:

```scala
abstract class OptionalQueryParamDecoderMatcher[T: QueryParamDecoder](name: String) {
  def unapply(params: Map[String, collection.Seq[String]]): Option[Option[T]] =
    params
      .get(name)
      .flatMap(_.headOption)
      .traverse(s => QueryParamDecoder[T].decode(QueryParameterValue(s)))
      .toOption
}
```

In that way, in the routes, you'll get a `Option[ValidationBool]` when using `+& ValidationParam(booleanValidated)`. 

Said that, I'd use `Boolean` instead:

```scala
object ValidationParam extends OptionalQueryParamDecoderMatcher[Boolean]("validation")
```